spec-0007/index.md

```diff
@@ -5,69 +5,69 @@ author:
   - "Stéfan van der Walt <stefanv@berkeley.edu>"
   - "Sebastian Berg <sebastianb@nvidia.com>"
   - "Pamphile Roy <roy.pamphile@gmail.com>"
+  - "Matt Haberland <mhaberla@calpoly.edu>"
   - Other participants in the discussion <not.yet@named.org>"
 discussion: https://github.com/scipy/scipy/issues/14322
 endorsed-by:
 ---
 
 ## Description
 
-<!--
-Briefly and clearly describe the proposal.
-Explain the general need and the advantages of this specific proposal.
-If relevant, include examples of how the new functionality would be used,
-intended use-cases, and pseudo-code illustrating its use.
--->
-
-There is disparity in the APIs libraries provide to seed random number generation.
-This SPEC suggests a single, pragmatic API for the ecosystem, taking into account technical and historical factors.
+Currently, libraries across the ecosystem provide various APIs for seeding random number generation.
+This SPEC suggests a unified, pragmatic API, taking into account technical and historical factors.
 Adopting such a uniform API will simplify the user experience, especially for those who rely on multiple projects.
 
-We recommend to:
+We recommend:
+
+- avoiding the use of global state and legacy bitstream generators, and
+- standardizing the usage and interpretation of an `rng` keyword for seeding.
 
-- Standardize the usage and interpretation of an `rng` keyword for seeding, and
-- avoid the use of global state and legacy bitstream generators.
+We suggest implementing these principles by:
 
-We suggest doing this by:
+- deprecating the use of `numpy.random.seed` to control the random state,
+- deprecating the use of `random_state`/`seed` arguments in favor of an `rng` argument in all functions that require random number generation, and
+- using `numpy.random.default_rng` to validate the `rng` argument and instantiate a `Generator`.
 
-- Passing the `rng` argument to `numpy.random.default_rng` to instantiate a `Generator`, and
-- deprecating the use of `np.seed` and `RandomState`.
+### Scope
+
+This is intended as a recommendation to all libraries that allow users to control the
+state of a NumPy random number generator. It is specifically targeted toward functions
+that allow use of `numpy.random.seed` to control the random state and accept
+`RandomState` instances via an argument with a name other than `rng`, but the ideas
+are more broadly applicable. Use of random number generators other than those provided
+by NumPy are not considered at this time, but may be addressed in a future version.
```

Comment:
I suspect we won't go into that in the future; this is mainly about saying "use `rng` to set a seed/provide a generator". If projects want to allow foreign generators, that's probably OK, as long as they use the same argument name.

Reply:
Haha OK, but that would not be strictly compatible with this as written because this specifies that `rng` would be validated with `default_rng`, which doesn't accept foreign Generators.